src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -155,32 +182,92 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
     @Override
     protected Position handlePosition(Position position) {
 
-        String url;
-        if (json) {
-            url = this.url;
-        } else {
-            try {
-                url = formatRequest(position);
-            } catch (UnsupportedEncodingException | JsonProcessingException e) {
-                throw new RuntimeException("Forwarding formatting error", e);
+        class AsyncRequestAndCallback implements InvocationCallback<Response> {
+
+            private int retries = 0;
+            private Map<String, Object> payload;
+            private Invocation.Builder requestBuilder;
+
+            AsyncRequestAndCallback(Position position) {
+
+                String formattedUrl;
+                try {
+                    formattedUrl = json ? url : formatRequest(position);
+                } catch (UnsupportedEncodingException | JsonProcessingException e) {
+                    throw new RuntimeException("Forwarding formatting error", e);
+                }
+
+                requestBuilder = client.target(formattedUrl).request();
+                if (header != null && !header.isEmpty()) {
+                    for (String line: header.split("\\r?\\n")) {
+                        String[] values = line.split(":", 2);
+                        requestBuilder.header(values[0].trim(), values[1].trim());
+                    }
+                }
+
+                if (json) {
+                    payload = prepareJsonPayload(position);
+                }
+
+                deliveryPending.incrementAndGet();
+
+                send();
             }
-        }
 
-        Invocation.Builder requestBuilder = client.target(url).request();
+            private void send() {
+                if (json) {
+                    requestBuilder.async().post(Entity.json(payload), this);
+                } else {
+                    requestBuilder.async().get(this);
+                }
+            }
 
-        if (header != null && !header.isEmpty()) {
-            for (String line: header.split("\\r?\\n")) {
-                String[] values = line.split(":", 2);
-                requestBuilder.header(values[0].trim(), values[1].trim());
+            private void retry() {
+                boolean ok = false;
+                try {
```

Comment:
Why do we need tries everywhere? Where do you expect exceptions?

Reply:
I'm just ensuring `deliveryPending` gets decremented if anything "unexpected" happens. Do you think we don't really need it on the `retry` or `run` methods?

Reply:
I guess it makes sense. Is there a way to have a single block though instead of nesting?

Reply:
Maybe I'm wrong, but I think the overridden `run` method for the new `TimerTask` object, the one that will run asynchronously, is at a different context, so we are not actually nesting try blocks. Please correct me.
Other way could be to create a new class, override the run method there, then use a new object of that class here. Do you really think we need that kind of stuff here? If you prefer that approach, no problem, I'll code it. Thank you.

Reply:
Correct, but it's just hard to read when code is nested so much. I think the easiest option to clean this up is make your class extend `TimerTask`.

Reply:
Interesting. Do you mean to make `AsyncRequestAndCallback` to implement both, `InvocationCallback` and `TimerTask`, and then use the same object (`this`) as the first parameter to `newTimeout`? That should do the job. Thanks.

Reply:
Yes, `AsyncRequestAndCallback` to implement both.

Reply:
Great. Let's do it. Thanks.